src/base/checked_runner.ml

```diff
@@ -256,8 +256,11 @@ struct
           let old = Run_state.as_prover s in
           Run_state.set_as_prover s true ;
           let value =
-            As_prover.Provider.run p (Run_state.stack s) (get_value s)
-              (Run_state.handler s)
+            As_prover.Provider.run p (get_value s) (Run_state.handler s)
+            |> Option.value_exn
```

Comment:
If you do this in a match statement, then we'll only compute the error message when there's actually an error.